wbb/modules/admin.py

```diff
@@ -217,3 +225,175 @@ async def banme(_, message: Message):
+    chat_id = message.chat.id
+    from_user_id = message.from_user.id
+
+    if (await app.get_chat_member(chat_id,
+                                  BOT_ID)).can_promote_members:
+        if message.from_user.id in admins \
+                or message.from_user.id in SUDO:
+            if (await app.get_chat_member(chat_id,
+                                          from_user_id)).can_promote_members \
+                or (await app.get_chat_member(chat_id, from_user_id)).status \
+                == 'creator' \
+                    or message.from_user.id in SUDO:
+
+                if message.text != '/demote':
+                    username = message.text.replace('/demote', '')
+                    user_id = (await app.get_users(username)).id
+                    await message.chat.promote_member(
+                        user_id=user_id,
+                        can_change_info=False,
+                        can_invite_users=False,
+                        can_restrict_members=False,
+                        can_delete_messages=False,
+                        can_pin_messages=False,
+                        can_promote_members=False)
+                    await message.reply_text('Promoted!')
+
+                else:
+                    user_id = message.reply_to_message.from_user.id
+                    await message.chat.promote_member(
+                        user_id=user_id,
+                        can_change_info=False,
+                        can_invite_users=False,
+                        can_restrict_members=False,
+                        can_delete_messages=False,
+                        can_pin_messages=False,
+                        can_promote_members=False)
+                    await message.reply_text('Demoted!')
+
+            else:
+                await message.reply_text("Yeah, I Can See You're An Admin,"
+                                         + " But You Don't Have Permissions"
+                                         + " To Demote Someone.")
+        else:
+            await message.reply_text("You're Not An Admin, Want A Good Ban?")
+    else:
+        await message.reply_text("Well, Your Know What?, I'M NOT AN ADMIN!"
+                                 + " MAKE ME ADMIN!")
+
+# Pin Messages
+
+
+@app.on_message(cust_filter.command(commands=("pin")))
+async def pin(_, message: Message):
+    admins = await list_admins(message.chat.id)
+    chat_id = message.chat.id
+    from_user_id = message.from_user.id
+
+    if message.from_user.id in admins \
+            or message.from_user.id in SUDO:
+        if (await app.get_chat_member(chat_id,
+                                      from_user_id)).can_pin_messages \
+            or (await app.get_chat_member(chat_id, from_user_id)).status \
+            == 'creator' \
+                or message.from_user.id in SUDO:
+
+            await message.reply_to_message.pin()
+    else:
+        await message.reply_text("You're Not An Admin, Stop Spamming!")
+
```

Comment:
maybe you can change this `if message.from_user.id in admins or message.from_user.id in SUDO:` to something like `if allow_admin_action:` since you are using this quite common in this file.

and define such function to return True or False.

Reply:
doing

Reply:
just for the record, this may not be solved at least not in this PR